web/models/package.ex

```diff
@@ -169,4 +221,43 @@ defmodule HexWeb.Package do
   defp sort(query, nil) do
     query
   end
+
+  defp parse_search(_, _ \\ [])
+  defp parse_search("", [nil | keys]), do: Enum.reverse(keys)
+  defp parse_search("", keys), do: Enum.reverse(keys)
+  defp parse_search(search, keys) do
+    {filter, tail} = parse_key(search, keys)
+    tail
+    |> String.strip
+    |> parse_search(filter)
+  end
+
+  defp parse_key(string, search) do
+    {k, v} =
+      Regex.split(~r/:/, string, parts: 2)
+      |> List.to_tuple
+    {v, tail} =
+      parse_value(v)
+    {add_param({String.to_atom(k), v}, search), tail}
+  end
+
+  defp parse_value(_, _ \\ nil)
+  defp parse_value(<<"\"", tail :: binary>>, nil), do: parse_value(tail, "\"")
+  defp parse_value(search, nil), do: parse_value(search, " ")
+  defp parse_value(value, delim) do
+    case String.split(value, delim, parts: 2) do
+      [value] -> {value, ""}
+      [value, tail] -> {value, tail}
+    end
+  end
```

Comment:
You could also write it like you did, but remove the second argument and do the `String.split` in a `do_parse_value` function.